locations/spiders/cef_gb.py

```diff
@@ -1,30 +1,22 @@
-import json
-
 from scrapy.linkextractors import LinkExtractor
 from scrapy.spiders import CrawlSpider, Rule
 
 from locations.google_url import extract_google_position
-from locations.linked_data_parser import LinkedDataParser
+from locations.structured_data_spider import StructuredDataSpider
 
 
-class CefGBSpider(CrawlSpider):
+class CefGBSpider(CrawlSpider, StructuredDataSpider):
     name = "cef_gb"
     item_attributes = {"brand": "City Electrical Factors", "brand_wikidata": "Q116495226"}
     start_urls = ["https://www.cef.co.uk/stores/directory"]
     rules = [
         Rule(LinkExtractor(restrict_xpaths='//a[@rel="next"]')),
-        Rule(LinkExtractor(allow="/stores/", restrict_xpaths='//ul[@id="directory"]'), callback="parse"),
+        Rule(LinkExtractor(allow="/stores/", restrict_xpaths='//ul[@id="directory"]'), callback="parse_sd"),
     ]
     custom_settings = {"ROBOTSTXT_OBEY": False}
+    requires_proxy = True
 
-    def parse(self, response, **kwargs):
-        ld = response.xpath('//script[@type="application/ld+json"]/text()').get()
-        ld = ld.replace("&quot;", '"')
-        ld_item = json.loads(ld)
-
-        item = LinkedDataParser.parse_ld(ld_item)
-        item["ref"] = response.url
-
+    def post_process_item(self, item, response, ld_data):
         extract_google_position(item, response)
```

Comment:
Odd this bit didn't work.

Reply:
https://github.com/alltheplaces/alltheplaces/issues/10399

Reply:
This function will only look in links, not inside of JSON.T